tests/runtime_neutral/test_discoverable_wrapper_host_visibility.py

```
@@ -0,0 +1,318 @@
+    def _write_external_wrapper_ledger_path(self, target_root: Path, external_wrapper: Path) -> None:
+        ledger_path = target_root / ".vibeskills" / "install-ledger.json"
+        ledger = json.loads(ledger_path.read_text(encoding="utf-8"))
+        ledger["specialist_wrapper_paths"] = [str(external_wrapper)]
+        ledger["payload_summary"]["host_visible_entry_names"] = ["vibe"]
+        ledger["payload_summary"]["host_visible_entry_count"] = 1
+        ledger_path.write_text(json.dumps(ledger), encoding="utf-8")
+
+    def test_shell_check_reports_host_visible_discoverable_entries(self) -> None:
+        with tempfile.TemporaryDirectory() as tempdir:
+            target_root = Path(tempdir) / "codex-root"
+            subprocess.run(
```

Ruff (0.15.9)

[error] `subprocess` call: check for execution of untrusted input

(S603)